packages/framework/src/Component/FlashMessage/FlashBagProvider.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\FlashMessage;
+
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+class FlashBagProvider
+{
+    /**
+     * @var \Symfony\Component\HttpFoundation\Session\Session
+     */
+    protected Session $session;
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Session\Session $session
+     */
+    public function __construct(SessionInterface $session)
```

Comment:
Request stack also should not be used in constuctor as you can see in paid inspections in Symfony plugin: https://espend.de/phpstorm/plugin/symfony - but I did not dig why

Reply:
ah damn :smile: I will look into it :slightly_smiling_face:

Reply:
That's weird... I would really want to know the reason... Symfony docs mention the usage with requestStack in constructor https://symfony.com/doc/current/session.html#basic-usage 🤔

Reply:
oh I see it now... you should not call methods on request stack inside constructor. You can autowire it and pass to property

Reply:
Ok, I have rewritten it with `RequestStack` in 14d52e333. From the discussion below the blog article, I understand that the request stack will not be a service as well one day (when there will be, as Fabien wrote, "a better session management system"), but at the moment, it is really the recommended way of accessing the session. Regarding the paid inspections - it says the request stack must not be used in the constructor but I believe at the moment the simple initialization is ok (when you look closer at the example provided, the method is called on the request stack directly in the constructor, which is IMHO a bad pattern in most of the cases with any service because of possible side effects :slightly_smiling_face: )

Reply:
....and fixed tests in 4e7cfe3fc :smile: